### PR TITLE
Implemented strong warning for deleting working groups

### PR DESCRIPTION
**Pull request by NolanChan**
First draft: feedback please!
#273 

The message text needs to be worked on, but the functionality is shown in the gif. Uses the Sweet Alert gem.

*The working group is actually deleted, but the gif repeats too fast to show it is deleted*

ezgif com-gif-maker 1

Second Draft:
Internalization added
Converted to coffeescript

### CODE REVIEW

**Comment by phillipoertel** [comment]
Curious: Does it work with accented characters and Umlauts? Try working group names like Übersetzungen and blablaé...

**Comment by NolanChan** [comment]
Übersetzungen and blablaé seem to work for me

**Comment by dottorer** [comment]
This looks really slick (and so does the gif...). If it works, we could integrate it (and possible re-use in other spots later). 

Good stuff, Nolan!

**Comment by phil-monroe** [comment]
@NolanChan very slick! My only real concern is internationalization. Instead of hardcoding the strings, try using the [i18n-js](https://github.com/fnando/i18n-js) library to have the strings come from the internationalization pipeline we have setup.

Here is an example where it is used elsewhere:
https://github.com/lale-help/lale-help/blob/b201c244ffe13f5edbd37ae07a81108e8562e5fa/app/assets/javascripts/views/circles/admin.coffee#L21

**Comment by NolanChan** [comment]
Thanks for the example @phil-monroe! I was looking briefly for an example of Javascript internalization but I couldn't find one.

**Comment by NolanChan** [comment]
Updated to include internalization and coffeescript. The way I structured the internalization seemed a bit clunky, hopefully someone can point me in a cleaner direction if possible

**Comment by phil-monroe** [comment]
hmm.... after thinking about this some more, we definitely need to make this a bit more generic and using I18n in JS land should really only be for fetching defaults. What if we did something like this:

``` yaml
# config/locales/en.yml
en:
  confirm:
    default:
      title: Are you sure?
      text: That seems potentially dangerous. Are you sure you want to continue?
      confirmButtonText: Confirm
      cancelButtonText: Cancel
      completed: Let it be done
      cancelled: Good call
      confirmQuery: CONFIRM
      
    a:
      title: Remove all the A's
      confirmQuery: Delete 
```

``` ruby
# app/views/some/view.haml
= link_to "Dangerous", some_path, data: { confirm: t('confirm.a') }
```

``` coffeescript
$.rails.showConfirmDialog = (link) ->
  data = link.attr('data-confirm')
  
  # merge hashes for defaults - http://stackoverflow.com/questions/929776/merging-associative-arrays-javascript
  data = $.extend({}, I18n.t('confirm.defaults'), data);
  data.type = 'input'
  data.showCancelButton = true
  data.confirmButtonColor: '#DD6B55'
  data.closeOnConfirm: false
  data.closeOnCancel: false
  
  swal data, (inputValue) ->
    if inputValue == data.confirmQuery
      # Not sure what the `swal` function signature is, but I think this gives the gist of what I'm thinking
      swal data.completedText, data.completedConfirmedText, 'success', (inputValue) ->
        $.rails.confirmed(link)
    else
      swal data.cancelledText, data.cancelConfirmedText, 'error'
```

I have a feeling this will get somewhat kludgy for the strings that need merge variables, but that should be solvable with something like this:
``` ruby
# app/views/some/view.haml
= link_to "Dangerous", some_path, data: { confirm: t('confirm.a').merge(key_to_override: I18n.t('somestring', variable: working_group.name)) }
```
Not the prettiest, but I'm sure we could find some syntatic sugar to clean it up.

What do you guys think?

**Comment by NolanChan** [comment]
Wow, the refactor really cleaned up a lot of the code, thanks @phil-monroe ! I can get started on the refactor as soon as possible if we're good to go in this direction

**Comment by dottorer** [comment]
Are we agreed on this being the way to do it? Then we could have @NolanChan look at the refactor and see if we can merge it in. Sorry this causes some back and forth...

**Comment by NolanChan** [comment]
Hi @phil-monroe , I tried cleaning it up. Please let me know if more stuff needs to be updated. 
Also I didn't include a wordier description that's on #273 

> "Warning - deleting this work group will also delete all its included tasks, supplies and project. This step cannot be undone. Are you sure you want to proceed?"

But I assume that we may need to in the future

File: app/views/circle/admins/working_groups.slim [hunk 1]
```diff
@@ -14,9 +14,9 @@ section
           - group.admins.active.each do |user|
             li= user.name
 
-        td
-          = link_to t('.edit'), edit_circle_working_group_path(current_circle, group), class: 'button-primary'
-          = link_to t('.delete'), circle_working_group_path(current_circle, group), class: 'button', method: 'delete', data: { confirm: t('.confirm_delete_working_group') }
+        td.actions
+          a.edit href=edit_circle_working_group_path(current_circle, group)
+          = link_to "", circle_working_group_path(current_circle, group), class: 'delete', method: 'delete', data: { confirm: {type: t('confirm.working_group').merge(title: t('confirm.working_group.title', working_group: group.name)), working_group: group.name } }
```
Comment (phillipoertel) [inline app/views/circle/admins/working_groups.slim]: changing

`... t('confirm.working_group').merge(title: ...`

to: 

`... t('confirm.working_group', title: ...`

should fix the problem.

**Comment by phillipoertel** [comment]
We tried to check the feature on the review app (https://lale-help-stg-pr-285.herokuapp.com, login with admin@lale.help / Example1234). Unfortunately the working group administration area of the circle doesn't work: https://lale-help-stg-pr-285.herokuapp.com/circles/2/admin/working_groups.

The error is: ActionView::Template::Error (undefined method merge for #<ActiveSupport::SafeBuffer:...>), the reason being that #merge is called on a translation string. See my comment on `app/views/circle/admins/working_groups.slim` for how to fix it.

**Comment by NolanChan** [comment]
The Working Group screen has been significantly changed enough that this code will not work anymore; sorry for missing the ball on this. I apologise for my slowness

**Comment by dottorer** [comment]
@NolanChan No worries, you brought us the Sweet Alert gem, which we will try to use in various places soon :-) Thanks for your work on Lale!

**Pull request closed without merging**